runtime/AK/Set.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022, Mustafa Quraish <mustafa@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "AK/HashTable.h"
+#include <initializer_list>
+
+namespace AK {
+
+template<typename T, typename TraitsForT, bool IsOrdered>
+class Set : public HashTable<T, TraitsForT, IsOrdered> {
+private:
+    using HashTableType = HashTable<T, TraitsForT, IsOrdered>;
+
+public:
+    using HashTableType::HashTable;
+
+    Set(std::initializer_list<T> list)
+    {
+        MUST(ensure_capacity(list.size()));
+        for (auto& item : list)
+            MUST(add(item));
+    }
+
+    using HashTableType::contains;
+
+    ErrorOr<HashSetResult> add(const T& value) { return HashTableType::try_set(value); }
```

Comment:
Please run clang-format on this, which will format as east const (needs to be added to CI...)

Reply:
Hmm, I did run clang-format on this, it doesn't seem to want to make it east-const. I'll do it manually.

Reply:
Pushed the fix.